tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/statements/vardeclr/VarDeclrSemanticTest.java

```diff
@@ -67,14 +69,19 @@ public void testIncompleteListenerDecl2() {
                       17, 1);
         validateError(result, indx++, "missing object keyword", 18, 1);
         validateError(result, indx++, "missing open brace token", 18, 1);
+        validateError(result, indx++, "missing object keyword", 19, 23);
```

Comment:
@lochana-chathura @rdulmina are these duplicate errors expected?

The source here is 

```ballerina
// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
//
// WSO2 Inc. licenses this file to you under the Apache License,
// Version 2.0 (the "License"); you may not use this file except
// in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.

listener

public function bar() {

}
```

Please note L78/L79 too.

Reply:
this the recovery for above code segment.
```bal
listener
 MISSING[object] MISSING[{]
public function bar()  MISSING[;] MISSING[object]{

}
 MISSING[] MISSING[;] MISSING[}] MISSING[] MISSING[=] MISSING[] MISSING[;]
```
they are not duplicate errors.

Reply:
They are not duplicate errors. Parser recovers another object field which is again an object type. Ideally parser should not recover like that. Can we create a parser issue for that?

Reply:
yea. we can give it a try to do better recovery here. +1 to create an issue.